### ISSUE TITLE
User reports difficulty setting up redirects in Emissary 2.0

### ISSUE THREAD

**Issue by efunk** [emissary-ingress/emissary#3709]
From user Tim Rice: 

Hi! Trying out the emissary-ingress 2.0 dev preview... having trouble getting redirects working, even using the basic redirect example from the docs

```
apiVersion: x.getambassador.io/v3alpha1
kind: AmbassadorMapping
metadata:
  name: redirect
spec:
  prefix: /redirect/
  service: httpbin.org
  host_redirect: true
  hostname: ‘*’
```

just returns a 404.
accessing an application in the cluster works just fine...
is this a known problem or are we doing something wrong?

Comment from Cindy:
The user is reporting an issue with redirects for Emissary 2.0 and it seems like he's using the new CRDs... he says it works for 1.13.

**Comment by kflynn** [comment]
@cindymullins-dw, can you check if the user has configured an `AmbassadorListener` and `AmbassadorHost`? cf https://www.getambassador.io/docs/emissary/2.0/howtos/configure-communications/

**Comment by timricese** [comment]
> @cindymullins-dw, can you check if the user has configured an `AmbassadorListener` and `AmbassadorHost`? cf https://www.getambassador.io/docs/emissary/2.0/howtos/configure-communications/

Hi! Im the user.

We had `AmbassadorListener` and `AmbassadorHost` configured. A "regular" mapping exposing a service in the cluster worked fine, however host_redirect did not.

**Comment by ChrisGitter** [comment]
I'm also unable to set up redirects.

My Setup:
- 2 Domains pointing at the server (example.de and example.com)
- 2 Listeners, one for HTTP and one for HTTPS
- 2 Hosts, one for each domain, without any wildcards or subdomains
- TLS certificates for both domains
- a service to serve on the domains

If I create two mappings without any redirects it all works fine and I can reach the service on both example.de and example.com. But if I want to redirect one domain to the other I get a 404 on the redirected domain. Here are the mappings:

```yml
apiVersion: getambassador.io/v3alpha1
kind:  Mapping
metadata:
  name:  example-service-mapping-de
spec:
  prefix: /
  service: example-service
  hostname: example.de
---
apiVersion: getambassador.io/v3alpha1
kind:  Mapping
metadata:
  name:  example-service-mapping-com
spec:
  prefix: /
  service: example.de
  hostname: example.com
  host_redirect: true
```

**Comment by Elvander** [comment]
Same issue here with setting up the redirect

`
apiVersion: getambassador.io/v3alpha1
kind: Mapping
metadata:
  name: redirect-singapore
  namespace: production-website
spec:
  hostname: '*'
  prefix: /
  service: example.co.nz
  host_redirect: true
`

**Comment by Elvander** [comment]
Same issue here

```
apiVersion: getambassador.io/v3alpha1
kind: Mapping
metadata:
  name: redirect-singapore
  namespace: production-website
spec:
  hostname: '*'
  prefix: /
  service: example.co.nz
  host_redirect: true
```

The result is a 404 instead of the expected 301 ...

**Comment by pjediny** [comment]
I see similar behavior in v1.14.2

**Comment by vjdhama** [comment]
As a workaround, I am using an `externalName` service.

**Comment by MattMencel** [comment]
@vjdhama can you explain how you used `externalName` mapping to resolve your redirect problem? I created an `externalName` service and pointed the redirect mapping to that but still get the `404` error.

The bugfix looks like it will be in 2.1.2, anyone have an idea on the timeframe for that release?

**Comment by khussey** [comment]
This was fixed in Emissary-ingresss 2.1.2, which is now available.
